### DISCUSSION TITLE
Home Assistant and Update Questions

### DISCUSSION THREAD

**Discussion by KyleStilkey** [gtxaspec/wz_mini_hacks#332] [Q&A]
Hi there!

I've recently installed this on a few of my Wyze V3's and it's working just fine. I just had a few questions to ask about updating the mini hack and Home Assistant.

So first with the update, is there a way to update the mini hack without having to take out the SD and update it every time? Is it done through SSH and if it can be I am a complete newb when it comes to that, I think I set it up right and SSH connected once but I have 0 idea what I am really doing at this point.

And with Home Assistant, I have made these work and added them as generic cameras I see video no problem but have 0 audio, is there something I'm doing wrong, limitation on Home Assistant's side?

Thanks to anyone that's able to help and answer my questions!

**Comment by bittles** [comment]
> Hi there!
> 
> 
> 
> I've recently installed this on a few of my Wyze V3's and it's working just fine. I just had a few questions to ask about updating the mini hack and Home Assistant.
> 
> 
> 
> So first with the update, is there a way to update the mini hack without having to take out the SD and update it every time? Is it done through SSH and if it can be I am a complete newb when it comes to that, I think I set it up right and SSH connected once but I have 0 idea what I am really doing at this point.
> 
> 
> 
> And with Home Assistant, I have made these work and added them as generic cameras I see video no problem but have 0 audio, is there something I'm doing wrong, limitation on Home Assistant's side?
> 
> 
> 
> Thanks to anyone that's able to help and answer my questions!

You can upgrade them by ssh'ing in and typing '/opt/wz_mini/bin/upgrade-run.sh'. For audio it's probably not enabled on the camera. If you have web interface enabled on the camera you can enable audio on there. If not, ssh in and type 'nano /opt/wz_mini/wz_mini.conf' and change enable audio from false to true. You'll have to reboot to enable the settings. Can also enable the web interface in that file.

**Reply by KyleStilkey** [reply]
Ok I do have Audio enable in the config and no audio is in Home Assistance.

I believe that SSH is set up correctly but when I try connecting using the username root I get Authentication failed so I'm not sure if once I put a key in the SSH if the username is changed, I have not really used SSH. It's new and trying to read up on how it works I'm just unsure where I'm missing something.

**Reply by bittles** [reply]
Do you use Putty and/or Puttygen?

**Reply by KyleStilkey** [reply]
I used Windows built in one, I thought I had it working but tried Puttygen when redoing it and it connects but says it's failed to connect using the username root.

**Reply by bittles** [reply]
Putty is nice and easy enough to use.  Whenever you're setting up your sd card before you put it into your camera, use PuttyGen and hit generate to make a new key.  Hit save private key and save public key and store them somewhere on your computer.  You'll also see in the upper part of PuttyGen "Public key for pasting into OpenSSH authorized_keys file: " with the key that you just generated.  Copy everything in that text box.  Go into the files on the sd card and the folders /SD_ROOT/opt/wz_mini/etc/ssh and open the authorized_keys file with notepad or any text editor.  Paste what you copied from PuttyGen into there and save it. 

With the authorized_keys file set up on the sd card plug it into your wyze cam and boot.  Open up Putty.  Type your camera's ip address into "Host Name (or IP address)".  On the left in the Category pane select Data (under Connection).  In Auto-login username type in root.  Back in the Category pane expand the SSH section (also under Connection) and select Auth.  At the bottom of the Auth screen it has "Private key file for authentication:".  Browse to the private key you saved from Puttygen and select it.  No go back to the top of the Category pane and select Session (where you started) and type in whatever you want to call these settings saved for this wyze cam in the box under "Saved Sessions" and click save.  Now you can just double click on your saved settings here to connect to your camera and you'll automatically SSH in.

**Reply by KyleStilkey** [reply]
OK I solved the SSH issue, I thank you a bunch for walking me through this!

Now I'm just stuck in trying to figure out why no sound is coming through. I think it may be a limitation on HA with supported CodeC. Unless someone has some success getting sound to work I'm just going to say this is the issue probably as it works just fine through VLC streams.
